docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+
+Current `FybrikApplicationSpec` has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of `clusterName` and `workloadSelector`. Unfortunately, based on this information we can 
+implement only one label-based `podSelector`, which will allow incoming connections from pods in the namespace where the NP 
+instance is deployed and have the defined label. In order to restrict incoming connections to Fybrik modules, the NP
+instances should be deployed in teh same namespace where the modules are running. However, Fybrik separates user's 
+objects, e.g. workloads and Fybrik objects, e.g. modules. Therefore, the existing mechanism should be extended.
+
+### Suggestions:
+
+- Extend the `FybrikApplicationSpec.selector` element with an array of `WorkloadLocations`, this array will be an input 
+element for the Network Policy `from`element of the data plane entry module. 
+- The `WorkloadLocation` wil have the following structure: 
+
+```go
+type WorkloadLocation struct {
+    // This is a workload's pod label selector which selects user workload pods. This field follows standard label
+    // selector semantics; if present but empty, it selects all pods.
+    //
+    // If NamespaceSelector is also set, then the WorkloadLocation as a whole selects
+    // the workload pods matching WorkloadPodSelector in the namespaces selected by NamespaceSelector.
+    // Otherwise, it selects the pods matching WorkloadPodSelector in the Fybrik data plane entry point namespace.
+    // +optional
+	WorkloadPodSelector *metav1.LabelSelector `json:"workloadPodSelector,omitempty" protobuf:"bytes,1,opt,name=workloadPodSelector"`
+
+    // Selects namespaces using cluster-scoped labels. This field follows standard label
+    // selector semantics; if present but empty, it selects all namespaces.
+    //
+    // If WorkloadPodSelector is also set, then the WorkloadLocation as a whole selects
+    // the workload pods matching WorkloadLocation in the namespaces selected by NamespaceSelector.
+    // Otherwise, it selects all Pods in the namespaces selected by NamespaceSelector.
+	// IF you want to select a specific namespace based on its name, it can be done by selecting an immutable label 
+	// `kubernetes.io/metadata.name` which is automatically set by the Kubernetes control plane. The value of the label 
+	// is the namespace name.
+    // +optional
+    NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,2,opt,name=namespaceSelector"`
+
+    // IPBlock defines policy on a particular IPBlock. If this field is set then neither of the other fields can be.
+	// the structure of the IPBlock is defined at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ipblock-v1-networking-k8s-io
+    // +optional
+    IPBlock *IPBlock `json:"ipBlock,omitempty" protobuf:"bytes,3,rep,name=ipBlock"`
```

Comment:
See previous comments about IP based isolation.  We either need to be clear about when to use it or leave it out for now, and wait until we talk about addressing the relevant problem.